zmq.hpp

```diff
@@ -1053,10 +1058,12 @@ namespace zmq
 
         void add (zmq::socket_t &socket, short events, handler_t handler)
         {
+            if (!socket)
+                throw error_t ();
```

Comment:
errno must be set to EINVAL for consistency, or error_t must be created with an explicit error code of EINVAL. An appropriate ctor of error_t must be added.

Reply:
Thank you for review. I don't think that setting EINVAL in cpp wrapper is a good idea, we have exceptions already for carrying information and should not rely on errno but I get the point that someone might rely on errno after throw already. Updated a review to throw only if zmq call fails so passing handling of nullptr socket to zmq.

Reply:
The ctor of error_t reads errno. It is so far only intended to be thrown when there was a libzmq error.
As it is now, the exception will contain a random errno.

Reply:
Have you seen updated pull request?